relayer/chain/substrate/chain.go

```diff
@@ -30,21 +32,27 @@ func NewChain(config *Config) (*Chain, error) {
 	log := logrus.WithField("chain", Name)
 
 	// Generate keypair from secret
-	kp, err := sr25519.NewKeypairFromSeed(config.PrivateKey, "")
+	kpParachain, err := sr25519.NewKeypairFromSeed(config.Parachain.PrivateKey, "")
+	if err != nil {
+		return nil, err
+	}
+
+	kpRelaychain, err := sr25519.NewKeypairFromSeed(config.Relaychain.PrivateKey, "")
 	if err != nil {
 		return nil, err
 	}
 
 	return &Chain{
-		config:   config,
-		conn:     NewConnection(config.Endpoint, kp.AsKeyringPair(), log),
-		listener: nil,
-		writer:   nil,
-		log:      log,
+		config:    config,
+		conn:      NewConnection(config.Parachain.Endpoint, kpParachain.AsKeyringPair(), log),
+		relayconn: NewConnection(config.Relaychain.Endpoint, kpRelaychain.AsKeyringPair(), log),
+		listener:  nil,
+		writer:    nil,
+		log:       log,
 	}, nil
 }
 
-func (ch *Chain) SetReceiver(ethMessages <-chan []chain.Message, ethHeaders <-chan chain.Header) error {
+func (ch *Chain) SetReceiver(ethMessages <-chan []chain.Message, ethHeaders <-chan chain.Header, _ chan<- store.DatabaseCmd) error {
```

Comment:
not sure having the general SetReceiver interface contain the beefy channel makes sense, especially given here its not needed/used. may be worth thinking about an alternative design/interface

Reply:
Agreed. The `Chain` interface is currently outdated. The Ethereum, Relaychain, and Parachain writers/readers use different selections of the following channels:
- subMessages/ethMessages <-chan []chain.Message
- ethHeaders chan<- chain.Header
- beefyMessages chan<- store.DatabaseCmd

Could we just deprecate the `SetReceiver`/`SetSender` methods on the `Chain` interface? Then each abstraction can pass only the required args it needs. @vgeddes thoughts?

Reply:
Included this as part of issue https://github.com/Snowfork/polkadot-ethereum/issues/340